bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmMonitorHandler.java

```diff
@@ -0,0 +1,328 @@
+            updateState(CHANNEL_TRIGGER_ALARM, m.isAlarm() ? OnOffType.ON : OnOffType.OFF);
+        }
+        Event event = m.getLastEvent();
+        if (event == null) {
+            clearEventChannels();
+        } else if (event.getEnd() != null) {
+            // If end is null, assume event hasn't completed yet
+            logger.trace("Monitor {}: Id:{}, Frames:{}, AlarmFrames:{}, Length:{}", m.getId(), event.getId(),
+                    event.getFrames(), event.getAlarmFrames(), event.getLength());
+            updateState(CHANNEL_EVENT_ID, new StringType(event.getId()));
+            updateState(CHANNEL_EVENT_NAME, new StringType(event.getName()));
+            updateState(CHANNEL_EVENT_CAUSE, new StringType(event.getCause()));
+            updateState(CHANNEL_EVENT_NOTES, new StringType(event.getNotes()));
+            updateState(CHANNEL_EVENT_START, new DateTimeType(
+                    ZonedDateTime.ofInstant(event.getStart().toInstant(), timeZoneProvider.getTimeZone())));
+            updateState(CHANNEL_EVENT_END, new DateTimeType(
+                    ZonedDateTime.ofInstant(event.getEnd().toInstant(), timeZoneProvider.getTimeZone())));
+            updateState(CHANNEL_EVENT_FRAMES, new DecimalType(event.getFrames()));
+            updateState(CHANNEL_EVENT_ALARM_FRAMES, new DecimalType(event.getAlarmFrames()));
+            updateState(CHANNEL_EVENT_LENGTH, new QuantityType<Time>(event.getLength(), SmartHomeUnits.SECOND));
+        }
+    }
+
+    private void clearEventChannels() {
+        updateState(CHANNEL_EVENT_ID, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_NAME, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_CAUSE, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_NOTES, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_START, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_END, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_FRAMES, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_ALARM_FRAMES, UnDefType.NULL);
+        updateState(CHANNEL_EVENT_LENGTH, UnDefType.NULL);
+    }
+
+    private void refreshImage() {
+        if (isLinked(CHANNEL_IMAGE)) {
+            scheduler.execute(this::getImage);
+        } else {
+            logger.trace("Monitor {}: Can't update image because '{}' channel is not linked", CHANNEL_IMAGE, monitorId);
+        }
+    }
+
+    private void getImage() {
+        try {
+            ZmBridgeHandler localHandler = bridgeHandler;
+            if (localHandler != null) {
+                logger.debug("Monitor {}: Updating image channel", monitorId);
+                RawType image = localHandler.getImage(monitorId, imageRefreshInterval);
+                updateState(CHANNEL_IMAGE, image != null ? image : UnDefType.UNDEF);
+            }
+        } catch (RuntimeException e) {
+            logger.debug("Monitor {}: Refresh image job got exception: {}", monitorId, e.getMessage(), e);
+        }
+    }
+
+    private void updateMonitorStatus() {
```

Comment:
See above

Reply:
This is called from `initialize`, so I think it's needed to properly set the handler status upon completion of `initialize`.

Reply:
Ok, if updating the child handlers is not necessary, I think this could be replaced by a one-liner, setting the state to online.

Reply:
With above insight, the whole method could replaced by `updateStatus(ThingStatus.ONLINE)`.

Reply:
Changed